arch/lkl/kernel/threads.c

```diff
@@ -123,7 +123,7 @@ struct task_struct *__switch_to(struct task_struct *prev,
 	}
 
 	if (_prev->dead) {
-		__sync_fetch_and_sub(&threads_counter, 1);
+		lkl__sync_fetch_and_sub((int *)&threads_counter, 1);
```

Comment:
Its probably better to just convert threads_counter to long instead of casting here.